src/vs/editor/common/codecs/simpleCodec/simpleDecoder.ts

```diff
@@ -0,0 +1,100 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { FormFeed } from './tokens/formFeed.js';
+import { Tab } from '../simpleCodec/tokens/tab.js';
+import { Word } from '../simpleCodec/tokens/word.js';
+import { VerticalTab } from './tokens/verticalTab.js';
+import { Space } from '../simpleCodec/tokens/space.js';
+import { NewLine } from '../linesCodec/tokens/newLine.js';
+import { VSBuffer } from '../../../../base/common/buffer.js';
+import { ReadableStream } from '../../../../base/common/stream.js';
+import { CarriageReturn } from '../linesCodec/tokens/carriageReturn.js';
+import { LinesDecoder, TLineToken } from '../linesCodec/linesDecoder.js';
+import { BaseDecoder } from '../../../../base/common/codecs/baseDecoder.js';
+
+/**
+ * A token type that this decoder can handle.
+ */
+export type TSimpleToken = Word | Space | Tab | VerticalTab | NewLine | FormFeed | CarriageReturn;
+
+/**
+ * Characters that stop a "word" sequence.
+ * Note! the `\r` and `\n` are excluded from the list because this decoder based on `LinesDecoder` which
+ * 	     already handles the `carriagereturn`/`newline` cases and emits lines that don't contain them.
+ */
+const STOP_CHARACTERS = [Space.symbol, Tab.symbol, VerticalTab.symbol, FormFeed.symbol];
+
+/**
+ * A decoder that can decode a stream of `Line`s into a stream
+ * of simple token, - `Word`, `Space`, `Tab`, `NewLine`, etc.
+ */
+export class SimpleDecoder extends BaseDecoder<TSimpleToken, TLineToken> {
+	constructor(
+		stream: ReadableStream<VSBuffer>,
+	) {
+		super(new LinesDecoder(stream));
+	}
+
+	protected override onStreamData(token: TLineToken): void {
+		// re-emit new line tokens
+		if (token instanceof CarriageReturn || token instanceof NewLine) {
+			this._onData.fire(token);
+
+			return;
+		}
+
+		// loop through the text separating it into `Word` and `Space` tokens
+		let i = 0;
+		while (i < token.text.length) {
+			// index is 0-based, but column numbers are 1-based
+			const columnNumber = i + 1;
+
+			// if a space character, emit a `Space` token and continue
+			if (token.text[i] === Space.symbol) {
+				this._onData.fire(Space.newOnLine(token, columnNumber));
+
+				i++;
+				continue;
+			}
+
+			// if a tab character, emit a `Tab` token and continue
+			if (token.text[i] === Tab.symbol) {
+				this._onData.fire(Tab.newOnLine(token, columnNumber));
+
+				i++;
+				continue;
+			}
+
+			// if a vertical tab character, emit a `VerticalTab` token and continue
+			if (token.text[i] === VerticalTab.symbol) {
```

Comment:
A vertical tab could be inside a file(or other medium) that we'd parse.

Reply:
Why does there need to be specific code to handle this? It doesn't seem to be handled in a particular way

Reply:
Not sure what do you mean? It is a separate distinct token that we care about.

Reply:
In what way do we care about it? How should the user use it in their file? Just trying to understand

Reply:
We care about it right now because it is one of the common `stop characters` for parsing out a sequence of characters for a prompt token, for instance a `#file:{path}` cannot contain a vertical tab character inside of it.